src/Parser/Core/Modules/Spells/BFA/AzeriteTraits/CrystallineCarapace.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+import Analyzer from 'Parser/Core/Analyzer';
+import { calculateAzeriteEffects } from 'common/stats';
+import SPELLS from 'common/SPELLS';
+
+export function crystallineCarapaceStats(combatant) {
+  if(!combatant.hasTrait(SPELLS.CRYSTALLINE_CARAPACE.id)) {
```

Comment:
You should only check for the trait in the `constructor` of the module, and set `this.active` to false if the player doesn't have the required trait.